bedrock/cms/wagtail_localize_smartling/callbacks.py

```diff
@@ -0,0 +1,54 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+# This file contains callback functions to be called by
+# github.com/mozilla/wagtail-localize-smartling/ and
+# hooked in via settings.
+#
+# The README.md of that repo covers what inputs are
+# provided and what outputs are needed
+
+from typing import TYPE_CHECKING
+
+from django.test import RequestFactory
+
+if TYPE_CHECKING:
+    from wagtail.models import Page
+    from wagtail_localize_smartling.models import Job
+
+from wagtaildraftsharing.models import WagtaildraftsharingLink
+from wagtaildraftsharing.views import SharingLinkView
+
+
+def _get_html_for_sharing_link(sharing_link: WagtaildraftsharingLink) -> str:
+    request = RequestFactory().get(sharing_link.url)
+    view_func = SharingLinkView.as_view()
+    resp = view_func(
+        request=request,
+        key=sharing_link.key,
+    )
+    return resp.content.decode("utf-8")
+
+
+def _get_full_url_for_sharing_link(sharing_link: WagtaildraftsharingLink, page: "Page") -> str:
+    return f"{page.get_site().root_url}{sharing_link.url}"
+
+
+def visual_context(smartling_job: "Job") -> list[str, str]:
```

Comment:
Should this be `-> tuple[str, str]:`? Not sure if mypy would complain but one is immutable and one isn't so it seems like it should.

Reply:
Yeah it should be a tuple, you're right!